### PR TITLE
Feature/uttmm 43 CRUD operation with projectTypes 

### PR DESCRIPTION
**Pull request by PanasiukAndriy**

### CODE REVIEW

File: TheraLang.DLL/Services/ProjectTypeService.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                throw new Exception($"Error when remove ProjectType by {nameof(ProjectType.Id)}: {projectTypeId}: ", ex);
+            }
+        }
+
+        public async Task Update(ProjectType projectType, int id)
+        {
+            try
+            {
+                projectType.UpdatedDateUtc = DateTime.UtcNow;
+                projectType.UpdatedById = id;
```
Comment (dgzargo) [inline TheraLang.DLL/Services/ProjectTypeService.cs]: remove plz! That's Pasha's task

File: TheraLang.DLL/Services/ProjectTypeService.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+                await _uow.SaveChangesAsync();
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Error when remove ProjectType by {nameof(ProjectType.Id)}: {projectTypeId}: ", ex);
+            }
+        }
+
+        public async Task Update(ProjectType projectType, int id)
```
Comment (dgzargo) [inline TheraLang.DLL/Services/ProjectTypeService.cs]:
```suggestion
        public async Task Update(ProjectType projectType)
```

File: TheraLang.Web/ClientApp/src/app/shared/api-endpoint.constants.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 export const liqpayCheckoutUrl = "https://www.liqpay.ua/api/3/checkout";
-export const baseUrl = "/api/";
+export const baseUrl = "http://localhost:5000/api/";
```
Comment (dgzargo) [inline TheraLang.Web/ClientApp/src/app/shared/api-endpoint.constants.ts]: revert this change

File: TheraLang.Web/ClientApp/src/app/project-info/resources-table-for-project/project-type/project-type-newHttp.service.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    delete(projectTypeId: number): Observable<any> {
+        return this.http.delete(projectTypeUrl + '/' + projectTypeId, { observe: "response" });
+    }
+
+    post(newProjectType: ProjectType): Observable<any> {
+        return this.http.post(projectTypeUrl, newProjectType);
+    }
+}
```
Comment (AlexBoichuk) [inline TheraLang.Web/ClientApp/src/app/project-info/resources-table-for-project/project-type/project-type-newHttp.service.ts]: You already have 'project-type-Http.service' with the same methods

File: TheraLang.Web/ClientApp/src/app/project-info/resources-table-for-project/project-type/project-type-Http.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { ProjectType } from './project-type.model';
+import { Observable } from 'rxjs';
+import { projectTypeUrl } from 'src/app/shared/api-endpoint.constants';
+
+@Injectable()
+export class ProjectTypeHttp {
+    constructor(private http: HttpClient) { }
+
+    getAllProjectTypes() {
+        return this.http.get(projectTypeUrl + '/');
```
Comment (AlexBoichuk) [inline TheraLang.Web/ClientApp/src/app/project-info/resources-table-for-project/project-type/project-type-Http.service.ts]:
```suggestion
You don't need ' / '
        return this.http.get(projectTypeUrl);
```

File: TheraLang.Web/ClientApp/src/app/project/http.service.ts [hunk 6]
```diff
@@ -48,9 +48,9 @@ export class HttpService {
         return this.http.put(this.url + '/' + project.id, project, { observe: 'response' });
     }
 
-    getAllProjectTypes() {
-        return this.http.get(this.url + '/' + 'projectTypes');
-    }
+    // getAllProjectTypes() {
+    //     return this.http.get(this.url + 'projectTypes');
+    // }
```
Comment (AlexBoichuk) [inline TheraLang.Web/ClientApp/src/app/project/http.service.ts]: If you don't need this method, delete it.

File: TheraLang.Web/ClientApp/src/app/project/project.component.ts [hunk 7]
```diff
@@ -4,6 +4,9 @@ import { Project } from './project';
 import { ProjectFormComponent } from '../project-form/project-form.component';
 import { ProjectService } from './project.service';
 import { DialogService } from '../shared/services/dialog.service';
+import { MatDialog, MatDialogConfig } from '@angular/material';
+// import { ProjectTypeComponent } from '../project-type/project-type.component';
+// import { ProjectTypeService } from '../project-type/project-type.service';
```
Comment (AlexBoichuk) [inline TheraLang.Web/ClientApp/src/app/project/project.component.ts]: If you don't need it, delete.

File: TheraLang.Web/ClientApp/src/app/project/project.component.ts [hunk 8]
```diff
@@ -14,16 +17,22 @@ import { NotificationService } from '../shared//services/notification.service';
...
   projects: Project[];
 
-  constructor(private httpService: HttpService,
+  constructor(
+    private httpService: HttpService,
     private dialogService: DialogService,
     private service: ProjectService,
-    private notificationService: NotificationService) { }
+    private dialog: MatDialog,
+    // private typeProjectService: ProjectTypeService,
```
Comment (AlexBoichuk) [inline TheraLang.Web/ClientApp/src/app/project/project.component.ts]: If you don't need it, delete.
